### PR TITLE
Added Badge.count constructor

### PR DESCRIPTION
**Pull request by HansMuller**
Added a named `Badge.count` constructor as a convenience for the common case of creating a Badge whose label is an integer with 1-3 digits. Count values greater than 999 are displayed as '999+'.

Per a suggest from @darrenaustin - https://github.com/flutter/flutter/pull/114560#issuecomment-1305847554.

### CODE REVIEW

File: packages/flutter/test/material/badge_test.dart [hunk 1]
```diff
@@ -111,6 +111,63 @@ void main() {
...
+    late final ThemeData theme;
+
+    Widget buildFrame(int count) {
+      return MaterialApp(
+        theme: ThemeData.light(useMaterial3: true),
+        home: Align(
+          alignment: Alignment.topLeft,
+          child: Builder(
+            builder: (BuildContext context) {
+              // theme.textTtheme is updated when the MaterialApp is built.
```
Comment (darrenaustin) [inline packages/flutter/test/material/badge_test.dart:125]:
```suggestion
              // theme.textTheme is updated when the MaterialApp is built.
```

File: packages/flutter/test/material/badge_test.dart [hunk 2]
```diff
@@ -111,6 +111,63 @@ void main() {
     expect(box, paints..rrect(rrect: RRect.fromLTRBR(-8, -4, 12, 12, const Radius.circular(8)), color: theme.colorScheme.error));
   });
 
+  // Essentially the same as 'Large Badge defaults'
+  testWidgets('Badge.count', (WidgetTester tester) async {
+    late final ThemeData theme;
```
Comment (darrenaustin) [inline packages/flutter/test/material/badge_test.dart:116]: Couldn't this just be: 
```suggestion
    late final ThemeData theme = ThemeData.light(useMaterial3: true);
```

And then you wouldn't have to pull it from the context later in the build? You could use it for both the build method and the expect clauses. Or I am missing something with the `Builder` widget?

**Review by darrenaustin — APPROVED** [review]
LGTM. Just a question about the test. Thx!

File: packages/flutter/test/material/badge_test.dart (hunk 2 above)
Comment (HansMuller) [reply packages/flutter/test/material/badge_test.dart:116]: ThemeData.textTheme is updated lazily so the initial value of `ThemeData.light(useMaterial3: true).textTheme.labelSmall` isn't the final value.

**Pull request merged**